packages/vite/src/node/__tests__/dev.spec.ts

```diff
@@ -17,3 +19,27 @@ describe('resolveBuildEnvironmentOptions in dev', () => {
     expect(config.build.rollupOptions).not.toHaveProperty('input')
   })
 })
+
+test('runner dynamic import', async () => {
```

Comment:
There is a test for dynamic import already: https://github.com/vitejs/vite/blob/v6/environment-api/packages/vite/src/node/ssr/runtime/__tests__/fixtures/dynamic-import.js

I would rather we update it there

Reply:
Thanks for the pointer. Moving there makes sense :+1:

The same issue reproduces there, but I'm seeing a weird error like it shows up on annotation:

```
PrettyFormatPluginError: [module runner] Dynamic access of "import.meta.env" is not supported. Please, use "import.meta.env.$$typeof" instead.
```

Locally, I confirmed the reproduction by console.log like this:

```ts
    console.log(modules.static === modules.dynamicAbsoluteExtension); // true
    console.log(modules.static === modules.dynamicAbsoluteFull); // false
```

Reply:
This is because there is a test file that does `export const meta = import.meta`, and accessing `.env` causes it to throw an error (the VItest equality checker tries to access `$$typeof`)

Reply:
Oh right I remember. For now I changed it to `expect(modules.static === modules.dynamicAbsoluteFull).toBe(true)` so that the error is not confusing.